### ISSUE TITLE
TypeError on pandas.tslib.NaT

### ISSUE THREAD

**Issue by jdanbrown** [wolever/pprintpp#18]
Fails on numpy NaT (not a time):
```py
>>> import pandas

>>> print(pandas.tslib.NaT)
NaT

>>> pprint.pprint(pandas.tslib.NaT)
NaT

>>> pprintpp.pprint(pandas.tslib.NaT)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-151-8d099b04508e> in <module>()
----> 1 pprintpp.pprint(pandas.tslib.NaT)

/Users/danb/miniconda3/lib/python3.5/site-packages/pprintpp/__init__.py in pprint(object, stream, indent, width, depth)
    131     printer = PrettyPrinter(
    132         stream=stream, indent=indent, width=width, depth=depth)
--> 133     printer.pprint(object)
    134
    135 def pformat(object, indent=4, width=80, depth=None):

/Users/danb/miniconda3/lib/python3.5/site-packages/pprintpp/__init__.py in pprint(self, object, state)
    286     def pprint(self, object, state=None):
    287         state = state or self.get_default_state()
--> 288         self._format(object, state)
    289         state.write("\n")
    290

/Users/danb/miniconda3/lib/python3.5/site-packages/pprintpp/__init__.py in _format(self, object, state)
    389         opener_closer_empty = (
    390             self._open_close_empty.get(typ) or
--> 391             self._open_close_empty.get(r)
    392         )
    393         if opener_closer_empty is not None:

TypeError: unhashable type: 'instancemethod'
```

Does work on numpy NaN (not a number):
```py
>>> import numpy

>>> print(numpy.nan)
nan

>>> pprint.pprint(numpy.nan)
nan

>>> pprintpp.pprint(numpy.nan)
nan
```

**Comment by wolever** [comment]
Ah, thanks! I'd be happy to merge a patch if you want to provide one, otherwise I'll have one out in the next couple of days.

**Comment by jdanbrown** [comment]
Nah, go for it. No rush.

Thanks!

**Comment by wolever** [comment]
Well, this turns out to be a slightly more interesting issue than I'd expected.

The core issue is that the Python 3 `instancemethod` type isn't hashable. `instancemethod` is a type used exclusively by non-core C extensions to define instance methods, and for some reason the `instancemethod_hash` function is commented out: https://github.com/python/cpython/blob/c30098c8c6014f3340a369a31df9c74bdbacc269/Objects/classobject.c#L569

There's a bit more discussion here: http://stackoverflow.com/q/40876368/71522

I've got this fixed in ff87c57!

**Comment by wolever** [comment]
I've also cut a new release - 0.3.0 - which includes this fix.
